client/lib/plans/index.js

```diff
@@ -66,6 +68,28 @@ export function getSitePlanSlug( siteID ) {
 	return get( site, 'plan.product_slug' );
 }
 
+function canUpgradeToPlan( planKey, site ) {
+	const offer = site.jetpack ? JETPACK : WPCOM;
+	return get( plansList, [ planKey, 'availableIn' ], () => false )( offer );
+}
+
+export function getUpgradePlanSlugFromPath( path, siteID ) {
+	const site = siteID ? sitesList.getSite( siteID ) : sitesList.getSelectedSite();
+	return find( Object.keys( plansList ), planKey => (
+		( planKey === path || getPlanPath( planKey ) === path ) &&
+		canUpgradeToPlan( planKey, site )
+	) );
+}
+
+export function getPlanPath( plan ) {
+	return get( plansList, [ plan, 'getPathSlug' ], () => undefined )();
+}
+
+export const getCartItem = ( productSlug, isFreeTrial = false ) => ( {
```

Comment:
> It doesn't do much

I was hoping it would serve an informational purpose to guide how the cart object looks like, which is not easy to figure out, but now that we have those routes we can remove it.

I have also put it in place to tackle `lib/cart-values/cart-items` because we have splenid concoctions like these:
```
function personalPlan( slug, properties ) {
	return planItem( slug, properties.isFreeTrial );
}
function premiumPlan( slug, properties ) {
	return planItem( slug, properties.isFreeTrial );
}
function businessPlan( slug, properties ) {
	return planItem( slug, properties.isFreeTrial );
}
```

These 3 functions do precisely nothing.
But I totally agree with you, we dont need more helpers. Removed in 6e65cdfed6cc22af5939043f6061cf1f82fd07f6